### DISCUSSION TITLE
Difference between "sympifiable" and "Sympifyable" in docstring

### DISCUSSION THREAD

**Discussion by 1e9abhi1e10** [sympy/sympy#24489] [General]
I found these two words in the docstring, Is there any difference between them?

**Comment by sylee957** [comment]
I don't think that there should be a difference, but picking such vocabulary like `sympifiable` can be jargon.

**Comment by 1e9abhi1e10** [comment]
Then which one is right, can I replace any one with another or it doesn't require a change

**Comment by sylee957** [comment]
I would defer making any changes over them because there doesn't seem like having agreement of using precise notation for documenting the semantics of variables in the community

**Comment by oscarbenjamin** [comment]
I would say that sympifiable is correct.

**Comment by asmeurer** [comment]
From what I can tell, both types of spellings are used in English. "Acidifiable" is an example of the former and "enjoyable" is an example of the latter, although [most examples](https://en.wiktionary.org/wiki/Category:English_terms_suffixed_with_-able) are the -yable form. Some guides say that the last letter only changes when adding -able when it is an e. Most examples seem to be the -yable

With that being said, I also personally like "sympifiable" because the other form looks wrong to me (although to be fair, "sympify" itself is a horrible word that I wish SymPy never used). We probably should just pick one since it's likely this will become an actual type name if we move forward with type hints (see the discussion at #17945).

**Reply by oscarbenjamin** [reply]
Nothing is ever a hard rule in English but usually the `yable` form is used where the preceding letter is a vowel and `iable` follows a consonant e.g. from the ones starting with `a`:
```
allayable
alloyable
annoyable
assayable
accompaniable
acidifiable
alkalifiable
alliable
amplifiable
appliable
associable
```
Hence `simpifiable`

**Reply by asmeurer** [reply]
That sounds good to me. @1e9abhi1e10 it should be simple to fix this.

**Reply by 1e9abhi1e10** [reply]
> Nothing is ever a hard rule in English but usually the `yable` form is used where the preceding letter is a vowel and `iable` follows a consonant e.g. from the ones starting with `a`:
> 
> ```
> allayable
> alloyable
> annoyable
> assayable
> accompaniable
> acidifiable
> alkalifiable
> alliable
> amplifiable
> appliable
> associable
> ```
> 
> Hence `simpifiable`
This one  `sympifiable` ?

**Comment by sylee957** [comment]
As the changes in #24512 indicates, I'd rather suggest to eventually narrow down to more explicit type notations like 
`Basic`, `Expr`, or even further narrowed down to `Integer, Symbol` and this eliminates the usage of `sympifiable`

**Reply by oscarbenjamin** [reply]
Is it possible to have something like `Sympifiable[Expr]`? The point of Sympifiable is that it expects types that are not necessarily already SymPy expressions.

**Reply by sylee957** [reply]
I think that I only got from the numpydoc style guide that type
annotations could represent 'semantics' if python typing annotation is too narrow or too verbose for that.

https://numpydoc.readthedocs.io/en/latest/format.html#parameters

but does not need to mimic something like generic type if it isn't formal enough. So I'd allow some freedom.
Numpydoc uses something like `array_like` though, so we can use `integer_like`, `expr_like`.

**Reply by asmeurer** [reply]
I think that's a good idea. Almost every use of "sympifiable"/"sympifyable" is as a type in a numpydoc parameters section (and it's only used in the physics module). It's implicit that a function that accepts `Expr` should also accept `int`. It doesn't benefit users much to see "Sympifiable" as a type instead of just "Expr". Most users will have to look up what that means. And either way, we should be consistent throughout the library, either using one or the other in docstrings.

Outside of types, "sympifiable" is only used in a few places, like https://docs.sympy.org/dev/explanation/glossary.html#term-sympify. 

> Is it possible to have something like Sympifiable[Expr]? The point of Sympifiable is that it expects types that are not necessarily already SymPy expressions.

I'm assuming you mean for type hints. I think that's how getitem on a type hint works, but I'm not completely clear https://peps.python.org/pep-0695/. 

For docstrings, I think this is starting to get way too specific to be actually useful for users. And it has the potential to go off the deep end with completely unreadable types.
